dotCMS/src/main/java/com/dotmarketing/webdav/DotWebdavHelper.java

```diff
@@ -835,6 +836,12 @@ public void createResource(String resourceUri, boolean publish, User user) throw
 		}
 
 	}
+	
+	private java.io.File writeDataIfEmptyFile(Folder folder, String fileName, java.io.File fileData) throws IOException{
+		Logger.warn(this, "The file " + folder.getPath() + fileName + " that is trying to be uploaded is empty. A byte will be written to the file because empty files are not allowed in the system");
```

Comment:
Let's include the IF in this method as well:
`if(fileData.length() == 0 && !Config.getBooleanProperty("CONTENT_ALLOW_ZERO_LENGTH_FILES", false) `

and then in the first call (line 924) it would be :
```
if (HttpManager.request().getUserAgentHeader().contains("Cyberduck") {
writeDataIfEmptyFile(params..)
}
```
and the second (line 974) will only be 
`writeDataIfEmptyFile(params..)`